cluster/charts/universal-crossplane/templates/upbound-agent/_deployment-spec.tpl

```diff
@@ -0,0 +1,87 @@
+{{- define "agent-spec" -}}
+replicas: 1
+selector:
+  matchLabels:
+    {{- include "selectorLabelsAgent" . | nindent 8 }}
+template:
+  metadata:
+    labels:
+      {{- include "selectorLabelsAgent" . | nindent 10 }}
+  spec:
+    serviceAccountName: {{ template "agent-name" . }}
+    {{- if .Values.imagePullSecrets }}
+    imagePullSecrets:
+    {{- range $index, $secret := .Values.imagePullSecrets }}
+    - name: {{ $secret }}
+    {{- end }}
+    {{ end }}
+    containers:
+      - name: agent
+        image: "{{ .Values.agent.image.repository }}:{{ .Values.agent.image.tag }}"
+        args:
+        - agent
+        - --tls-cert-file
+        - /etc/certs/upbound-agent/tls.crt
+        - --tls-key-file
+        - /etc/certs/upbound-agent/tls.key
+        - --xgql-ca-bundle-file
+        - /etc/certs/upbound-agent/ca.crt
+        - --nats-endpoint
+        - nats://{{ .Values.upbound.connectHost }}:{{ .Values.upbound.connectPort | default "443" }}
+        - --upbound-api-endpoint
+        - {{ .Values.upbound.apiURL }}
+        - --pod-name
+        - $(POD_NAME)
+        - --control-plane-token
+        - $(CP_TOKEN)
+        {{- if .Values.agent.config.debugMode }}
```

Comment:
I noticed we are passing these as both flags and as env vars. `kong` allows you to specify env vars to populate flags, which would mean we would only need to pass those env vars and they would automatically be present in the `AgentCmd` struct we use, which would avoid some duplication.

Reply:
Good point, fixed.